test/hummingbot/connector/exchange/bitmart/test_bitmart_api_user_stream_data_source.py

```diff
@@ -216,61 +224,70 @@ def test_listening_process_logs_exception_during_events_subscription(self, ws_co
-        # Add the authentication response for the websocket
-        self.mocking_assistant.add_websocket_text_message(
-            ws_connect_mock.return_value,
-            json.dumps({"event": "login"}))
-
-        self.mocking_assistant.add_websocket_text_message(ws_connect_mock.return_value, json.dumps('dummyMessage'))
+        try:
+            self.listening_task = self.ev_loop.create_task(
+                self.data_source.listen_for_user_stream(self.ev_loop,
+                                                        messages))
+            # Add the authentication response for the websocket
+            self.mocking_assistant.add_websocket_text_message(
+                ws_connect_mock.return_value,
+                json.dumps({"event": "login"}))
+            # Add invalid json message
+            self.mocking_assistant.add_websocket_text_message(ws_connect_mock.return_value, 'invalid message')
+            self.ev_loop.run_until_complete(self.listening_task)
+        except Exception:
+            pass
 
-    def _raise_asyncio_timeout_exception(self):
-        raise asyncio.TimeoutError()
+        self.assertTrue(self._is_logged("ERROR", "Unexpected error when parsing BitMart user_stream message. "))
+        self.assertTrue(self._is_logged("ERROR", "Unexpected error with BitMart WebSocket connection. "
+                                                 "Retrying after 30 seconds..."))
 
-    def test_inner_messages_timeout_ping(self):
-        async def call_inner(ws):
-            async for _ in self.data_source._inner_messages(ws):
-                return
-        ping_messages = []
-        ws_connect_mock = self.mocking_assistant.create_websocket_mock()
-        ws_connect_mock.ping.side_effect = lambda: ping_messages.append("ping")
-        self.data_source.MESSAGE_TIMEOUT = 0.1
-        with self.assertRaises(asyncio.exceptions.TimeoutError):
-            asyncio.get_event_loop().run_until_complete(asyncio.wait_for(call_inner(ws_connect_mock), 0.2))
-        self.assertIn("ping", ping_messages)
-
-    def test_inner_messages_timeout_ping_timeout(self):
-        async def call_inner(ws):
-            async for _ in self.data_source._inner_messages(ws):
-                return
-        ws_connect_mock = self.mocking_assistant.create_websocket_mock()
-        messages = asyncio.Queue()
-        ws_connect_mock.ping.side_effect = self.mocking_assistant.async_partial(messages.get)
+    @patch("hummingbot.connector.exchange.bitmart.bitmart_api_user_stream_data_source.BitmartAPIUserStreamDataSource._authenticate")
+    @patch("hummingbot.connector.exchange.bitmart.bitmart_api_user_stream_data_source.BitmartAPIUserStreamDataSource._subscribe_to_channels")
+    @patch("hummingbot.connector.exchange.bitmart.bitmart_api_user_stream_data_source.BitmartAPIUserStreamDataSource._init_websocket_connection",
+           new_callable=AsyncMock)
+    def test_listen_for_user_stream_inner_messages_recv_timeout(self, ws_connect_mock, *_):
```

Comment:
I don't think mocking the internal logic of the user stream class is a good idea. We should be working based on mocking the http requests or the websocket interactions, as in the rest of the tests.

Reply:
I understand, this was a change I introduced, so I will remedy it.